### PR TITLE
virsh_setmaxmem: Add "readonly" test

### PR DESCRIPTION
**Pull request by hahakiki2010**
Signed-off-by: Yan Li <yannli@redhat.com>

### CODE REVIEW

**Comment by hahakiki2010** [comment]
Test passed
(virtenv_py3) [root@localhost tp-libvirt]# avocado run --vt-type libvirt virsh.setmaxmem.readonly_test.running.domname.half_mem.config
JOB ID     : 6cc627358c2e9e69b4c0a6bba75865a4985a3f38
JOB LOG    : /root/avocado/job-results/job-2018-07-25T14.55-6cc6273/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.setmaxmem.readonly_test.running.domname.half_mem.config: PASS (4.98 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 16.01 s

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_setmaxmem.cfg [hunk 1]
```diff
@@ -88,3 +88,18 @@
...
+                    setmaxmem_flags = "--config"
+            variants:
+                - half_mem:
+                    setmaxmem_mem_ref = "halfless"
+            variants:
+                - domname:
+                    setmaxmem_vm_ref = "domname"
+            variants:
+                - running:
+                    start_vm = "yes"
```
Comment (kylazhang) [inline libvirt/tests/cfg/virsh_cmd/domain/virsh_setmaxmem.cfg]: why use Cartesian here?

File: libvirt/tests/src/virsh_cmd/domain/virsh_setmaxmem.py [hunk 2]
```diff
@@ -145,6 +145,11 @@ def print_debug_stats(original_vmxml_mem, original_dominfo_mem,
     dargs = {'flagstr': flags,
              'use_kilobytes': use_kilobytes,
              'uri': uri, 'ignore_status': True, "debug": True}
+
+    readonly = ("yes" == params.get("setmaxmem_readonly", "no"))
+    if readonly:
+        dargs.update({"readonly": readonly})
+
```
Comment (kylazhang) [inline libvirt/tests/src/virsh_cmd/domain/virsh_setmaxmem.py]: I think we need readonly err msg check, else we can not make sure it raise the right error

**Comment by hahakiki2010** [comment]
duplicate with https://github.com/autotest/tp-libvirt/pull/1729

**Pull request closed without merging**